types/block.go

```diff
@@ -84,11 +80,6 @@ func (b *Block) ValidateBasic() error {
 		)
 	}
 
-	// NOTE: b.Data.Txs may be nil, but b.Data.Hash() still works fine.
-	if w, g := b.Data.Hash(), b.DataHash; !bytes.Equal(w, g) {
-		return fmt.Errorf("wrong Header.DataHash. Expected %X, got %X", w, g)
-	}
```

Comment:
[nit][not blocking]
could we add a comment in this function's docs that point out that this check has been moved to the app?

Reply:
> has been moved to the app

Specifically this? https://github.com/celestiaorg/celestia-app/blob/65d50da1cdda569bebe3d881c6da58ab59e58a90/app/process_proposal.go#L124-L129

Reply:
Resolved in https://github.com/celestiaorg/celestia-core/pull/850/commits/f43bc1548810ebd2ab2285fee2a511ac028453c6